## Importaciones

In [44]:
import pandas as pd 
import numpy as np 
import matplotlib.pyplot as plt

PRESA01 = pd.read_csv("datos/SA01_DATA.csv")
PRESA01.head()

GHO (CODE)                          GHO (DISPLAY)  \
0  #indicator+code                        #indicator+name   
1         WHS6_102  Hospital beds (per 10 000 population)   
2         WHS6_102  Hospital beds (per 10 000 population)   
3         WHS6_102  Hospital beds (per 10 000 population)   
4         WHS6_102  Hospital beds (per 10 000 population)   

                                           GHO (URL) YEAR (DISPLAY)  \
0                                     #indicator+url     #date+year   
1  https://www.who.int/data/gho/data/indicators/i...           2014   
2  https://www.who.int/data/gho/data/indicators/i...           2020   
3  https://www.who.int/data/gho/data/indicators/i...           2019   
4  https://www.who.int/data/gho/data/indicators/i...           2017   

          STARTYEAR         ENDYEAR REGION (CODE) REGION (DISPLAY)  \
0  #date+year+start  #date+year+end  #region+code     #region+name   
1              2014            2014           AMR         Americas   
2              2020            2020           AMR         Americas   
3              2019            2019           AMR         Americas   
4              2017            2017           AMR         Americas   

  COUNTRY (CODE) COUNTRY (DISPLAY) DIMENSION (TYPE) DIMENSION (CODE)  \
0  #country+code     #country+name  #dimension+type  #dimension+code   
1            COL          Colombia              NaN              NaN   
2            COL          Colombia              NaN              NaN   
3            COL          Colombia              NaN              NaN   
4            COL          Colombia              NaN              NaN   

  DIMENSION (NAME)               Numeric             Value  \
0  #dimension+name  #indicator+value+num  #indicator+value   
1              NaN              16.25071              16.3   
2              NaN                16.891              16.9   
3              NaN              17.42688              17.4   
4              NaN              17.32908              17.3   

                    Low                   High  
0  #indicator+value+low  #indicator+value+high  
1                   NaN                    NaN  
2                   NaN                    NaN  
3                   NaN                    NaN  
4                   NaN                    NaN

# Exploración Inicial de los Datos 🔎
Para analizar la información de mi dataset, su tamaño y las columnas. 

In [45]:
PRESA01.info()
PRESA01.shape

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 12 entries, 0 to 11
Data columns (total 17 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   GHO (CODE)         12 non-null     object
 1   GHO (DISPLAY)      12 non-null     object
 2   GHO (URL)          12 non-null     object
 3   YEAR (DISPLAY)     12 non-null     object
 4   STARTYEAR          12 non-null     object
 5   ENDYEAR            12 non-null     object
 6   REGION (CODE)      12 non-null     object
 7   REGION (DISPLAY)   12 non-null     object
 8   COUNTRY (CODE)     12 non-null     object
 9   COUNTRY (DISPLAY)  12 non-null     object
 10  DIMENSION (TYPE)   1 non-null      object
 11  DIMENSION (CODE)   1 non-null      object
 12  DIMENSION (NAME)   1 non-null      object
 13  Numeric            12 non-null     object
 14  Value              12 non-null     object
 15  Low                1 non-null      object
 16  High               1 non-null      object
dtyp

(12, 17)

In [46]:
PRESA01.head()

GHO (CODE)                          GHO (DISPLAY)  \
0  #indicator+code                        #indicator+name   
1         WHS6_102  Hospital beds (per 10 000 population)   
2         WHS6_102  Hospital beds (per 10 000 population)   
3         WHS6_102  Hospital beds (per 10 000 population)   
4         WHS6_102  Hospital beds (per 10 000 population)   

                                           GHO (URL) YEAR (DISPLAY)  \
0                                     #indicator+url     #date+year   
1  https://www.who.int/data/gho/data/indicators/i...           2014   
2  https://www.who.int/data/gho/data/indicators/i...           2020   
3  https://www.who.int/data/gho/data/indicators/i...           2019   
4  https://www.who.int/data/gho/data/indicators/i...           2017   

          STARTYEAR         ENDYEAR REGION (CODE) REGION (DISPLAY)  \
0  #date+year+start  #date+year+end  #region+code     #region+name   
1              2014            2014           AMR         Americas   
2              2020            2020           AMR         Americas   
3              2019            2019           AMR         Americas   
4              2017            2017           AMR         Americas   

  COUNTRY (CODE) COUNTRY (DISPLAY) DIMENSION (TYPE) DIMENSION (CODE)  \
0  #country+code     #country+name  #dimension+type  #dimension+code   
1            COL          Colombia              NaN              NaN   
2            COL          Colombia              NaN              NaN   
3            COL          Colombia              NaN              NaN   
4            COL          Colombia              NaN              NaN   

  DIMENSION (NAME)               Numeric             Value  \
0  #dimension+name  #indicator+value+num  #indicator+value   
1              NaN              16.25071              16.3   
2              NaN                16.891              16.9   
3              NaN              17.42688              17.4   
4              NaN              17.32908              17.3   

                    Low                   High  
0  #indicator+value+low  #indicator+value+high  
1                   NaN                    NaN  
2                   NaN                    NaN  
3                   NaN                    NaN  
4                   NaN                    NaN

### Información contenida en el dataset:
- **GHO (CODE)**: `WHS6_102` - Código del indicador (número de camas hospitalarias por cada 10,000 habitantes).
- **GHO (DISPLAY)**: `Hospital beds (per 10 000 population)` - Descripción del indicador.
- **YEAR (DISPLAY)**: `2014` - El año en que se reportó el dato.
- **STARTYEAR** y **ENDYEAR**: `2014` - El periodo de tiempo en el que se tomó la medición.
- **REGION (CODE)**: `AMR` - América (Américas).
- **COUNTRY (CODE)**: `COL` - Colombia.
- **Numeric**: `16.25071` - Valor numérico preciso de camas hospitalarias por cada 10,000 habitantes.
- **Value**: `16.3` - Valor redondeado del indicador.
- **Low** y **High**: `NaN` - No se proporcionan rangos en este caso.

### Buscando valores nulos

In [47]:
valoresnull = PRESA01.isnull().sum()
valoresnull


GHO (CODE)            0
GHO (DISPLAY)         0
GHO (URL)             0
YEAR (DISPLAY)        0
STARTYEAR             0
ENDYEAR               0
REGION (CODE)         0
REGION (DISPLAY)      0
COUNTRY (CODE)        0
COUNTRY (DISPLAY)     0
DIMENSION (TYPE)     11
DIMENSION (CODE)     11
DIMENSION (NAME)     11
Numeric               0
Value                 0
Low                  11
High                 11
dtype: int64

Para conocer el porcentaje de valores nulos por columna. 

In [48]:
nvalorespct = PRESA01.isnull().mean() * 100
nvalorespct

GHO (CODE)            0.000000
GHO (DISPLAY)         0.000000
GHO (URL)             0.000000
YEAR (DISPLAY)        0.000000
STARTYEAR             0.000000
ENDYEAR               0.000000
REGION (CODE)         0.000000
REGION (DISPLAY)      0.000000
COUNTRY (CODE)        0.000000
COUNTRY (DISPLAY)     0.000000
DIMENSION (TYPE)     91.666667
DIMENSION (CODE)     91.666667
DIMENSION (NAME)     91.666667
Numeric               0.000000
Value                 0.000000
Low                  91.666667
High                 91.666667
dtype: float64

In [49]:
- **"DIMENSION (TYPE)", "DIMENSION (CODE)", "DIMENSION (NAME)":** Tienen 91.67% de valores nulos, lo que significa que casi todo el contenido de estas columnas está vacío.
- **"Low" y "High":** Tienen un 91.67% de valores nulos. Esto sugiere que en la mayoría de los casos no se tiene información sobre el rango de valores, lo cual podría ser relevante dependiendo del tipo de análisis que quieras realizar. **Estas columnas están relacionadas entre sí.** Podemos observar que el porcentaje de **valores faltantes es el mismo** y aparte, el contenido e información que nos proporciona, se relaciona entre dichas columnas.
- Las demás columnas **NO** tienen valores nulos.

SyntaxError: invalid syntax (3718368042.py, line 1)

# Limpiando los Datos 🧹
Decidí no incluir las columnas:  
**"GHO (URL)" =** Es el url del código del indicador y no afecta en nada su eliminación.
**"DIMENSION (CODE)", "DIMENSION (NAME)", "Low", "High" =** Columnas relacionadas entre sí que tienen el 91% de los datos nulos. 
Y eliminé el index 0, ya que era texto de guía. 

In [ ]:
INDEX = PRESA01.drop(0)
SA01 = INDEX.drop(columns=["GHO (URL)", "DIMENSION (TYPE)",  "DIMENSION (CODE)", "DIMENSION (NAME)", "Low", "High"])
SA01.head(20)


# Analisis 
Número de camas por año, para analizar cómo ha incrementado. 

In [ ]:
año_camas = SA01.groupby("STARTYEAR")["Numeric"].sum().sort_values()
año_camas

plt.figure(figsize=(20, 6))
plt.scatter(año_camas.index, año_camas.values, c="purple")
plt.xlabel('Camas')
plt.ylabel('Año')
plt.title('Año vs Camas')
plt.show()

plt.figure(figsize=(20, 6))
plt.bar(año_camas.index, año_camas.values)
plt.show()
